src/main/java/utils/FileStream.java

```diff
@@ -22,67 +23,69 @@ public static Path getFileName(Path filePath, String ending) {
     }
 
     //create file by his name
-    public static FileWriter createFile(Path fileName) throws IOException {
-        //create the decrypted file
-        File file;
-        file = new File(fileName.toString());
-        if (!file.createNewFile()) {
-                PrintWriter writer = new PrintWriter(file);
-                writer.print("");
-                writer.close();
-            }
-
-        //create the file-writer for the file
-        return new FileWriter(file.getAbsoluteFile());
+    public static File createFile(Path fileName) throws IOException {
+       try {
+           //create the decrypted file
+           File file;
+           file = new File(fileName.toString());
+           if (!file.createNewFile()) {
+               PrintWriter writer = new PrintWriter(file);
+               writer.print("");
+               writer.close();
+           }
+
+           //create the file-writer for the file
+           return file;
+       }
+       catch (IOException ex) {
+           throw new IOException("The function can't succeed to create file. \n File: "+fileName.getFileName() +"\n Error: "+ex.getMessage());
+       }
     }
 
     //get the key file
-    public static List<Integer> getListOfIntegers(Path keyFilePath) throws FileNotFoundException {
-
-        //read the key from the file
+    public static List<Integer> getListOfIntegers(Path keyFilePath) throws FileNotFoundException, InvalidEncryptionKeyException {
         File keyFile = new File(keyFilePath.toString());
-        Scanner itemReader = new Scanner(keyFile);
-
-        //get the key from the file
-        List<Integer> items = new ArrayList<>();
-        while (itemReader.hasNextLine()) {
-            int item = Integer.parseInt(itemReader.nextLine());
-            items.add(item);
+        //read the key from the file
+        try(Scanner itemReader = new Scanner(keyFile)) {
+            //get the key from the file
+            List<Integer> items = new ArrayList<>();
+            while (itemReader.hasNextLine()) {
+                int item = Integer.parseInt(itemReader.nextLine());
+                items.add(item);
+            }
+            return items;
+        }catch (FileNotFoundException ex) {
+            throw new FileNotFoundException("The source file isn't found. \n File: "+keyFilePath.getFileName()+"\n Error:"+ ex.getMessage());
+        }
+        catch (NumberFormatException ex) {
+            throw new InvalidEncryptionKeyException("The item isn't a pure number.\n"+ex.getMessage());
         }
-
-        itemReader.close();
-        return items;
     }
 
     //get the  message from the file
     public static String getFileContent(Path filePath) throws FileNotFoundException {
```

Comment:
read might be more accurate and understable
```suggestion
    public static String readFileContent(Path filePath) throws FileNotFoundException {
```

Reply:
done